theano/sandbox/cuda/dnn.py

```diff
@@ -2246,9 +2247,13 @@ def local_dnn_conv_inplace(node):
         inputs = list(node.inputs)
         dest = inputs[2]
         if (dest.owner and
-                isinstance(dest.owner.op, GpuAllocEmpty) and
+                type(dest.owner.op) is GpuAllocEmpty and
```

Comment:
> `GpuAlloc` inherits from `GpuAllocEmpty`

Oh, I forgot. Wouldn't it be better to just swap the cases then, so we can still use `isinstance`?
```python
if ... and isinstance(..., GpuAlloc) and ...:
    ...
elif ... and isinstance(..., GpuAllocEmpty) and ...:
    ...
```
Directly checking for the type could be problematic if additional subclasses of GpuAlloc or GpuAllocEmpty are introduced.